src/embedded.ts

```diff
@@ -280,7 +299,80 @@ export class RoomWidgetClient extends MatrixClient {
         content: any,
         stateKey = "",
     ): Promise<ISendEventResponse> {
-        return await this.widgetApi.sendStateEvent(eventType, stateKey, content, roomId);
+        return this.validateSendEventResponse(
+            await this.widgetApi.sendStateEvent(eventType, stateKey, content, roomId),
+        );
+    }
+
+    /**
+     * @experimental This currently relies on an unstable MSC (MSC4140).
+     */
+    // eslint-disable-next-line
+    public async _unstable_sendStateFuture<K extends keyof StateEvents, F extends SendFutureRequestOpts>(
+        roomId: string,
+        futureOpts: F,
+        eventType: K,
+        content: StateEvents[K],
+        stateKey: string,
+    ): Promise<SendFutureResponse<F>>;
```

Comment:
This overload seems to not add anything?

Reply:
This type signature is needed to properly override the method from the parent class, and the one following this is needed to avoid some typechecking errors, as type narrowing of return types is finicky when conditional parameter types are involved.

But I can avoid using an overload if I simply cast the return value, so I'll go with that.